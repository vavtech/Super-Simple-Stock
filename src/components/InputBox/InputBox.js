import TextField from '@material-ui/core/TextField';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
    root: {
        '& > *': {
            margin: theme.spacing(1),
            width: '25ch',
        },
    },
}));


const InputBox = (props) => {

    const classes = useStyles();

    return (
        <div>
            <form className={classes.root} noValidate autoComplete="off">
                <TextField onChange={props.handler} id={props.id}
                    type="number"
                    label={props.label}
                    value={props.value}
                    variant="outlined" />
            </form>
        </div>
    );
}

export default InputBox;