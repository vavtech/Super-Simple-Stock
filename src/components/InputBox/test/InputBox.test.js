import React from 'react';
import InputBox from '../InputBox';
import Enzyme, { shallow } from 'enzyme';
import TextField from '@material-ui/core/TextField';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

Enzyme.configure({ adapter: new Adapter() })

describe('InputBox', () => {

    it('should be defined', () => {
        expect(InputBox).toBeDefined();
    });

    it('should render correctly', () => {
        const tree = shallow(
            <InputBox label='text' />
        );
        expect(tree).toMatchSnapshot();
    });

    it('should have a button value', () => {
        const tree = shallow(
            <InputBox label='text' />
        );
        expect(tree.find(TextField).text()).toBe('');
    });

});