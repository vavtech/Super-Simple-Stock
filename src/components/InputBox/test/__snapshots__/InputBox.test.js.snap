// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBox should render correctly 1`] = `ShallowWrapper {}`;
